wallet/wallet.go

```diff
@@ -2852,6 +2852,42 @@ func (w *Wallet) LockedOutpoints() []btcjson.TransactionInput {
 	return locked
 }
 
+// LeaseOutput locks an output to the given ID, preventing it from being
+// available for coin selection. The absolute time of the lock's expiration is
+// returned. The expiration of the lock can be extended by successive
+// invocations of this call.
+//
+// Outputs can be unlocked before their expiration through `UnlockOutput`.
+// Otherwise, they are unlocked lazily through calls which iterate through all
+// known outputs, e.g., `CalculateBalance`, `ListUnspent`.
+//
+// If the output is not known, ErrUnknownOutput is returned. If the output has
+// already been locked to a different ID, then ErrOutputAlreadyLocked is
+// returned.
+//
+// NOTE: This differs from LockOutpoint in that outputs are locked for a limited
+// amount of time and their locks are persisted to disk.
+func (w *Wallet) LeaseOutput(id uint64, op wire.OutPoint) (time.Time, error) {
```

Comment:
These should also call `LockedOutpoint` so things are in-sync. `LockedOutputs` should also refer to this.

Reply:
I'm not sure that's safe? The lease lock will expire at some point, but that's not the case with `LockedOutpoint`, unless we detect when it has expired and remove it ourselves.

Reply:
Well, I guess it depends on the goals here. I had envisioned we would have a sort of unified abstraction for al of this, but it's also feasible that we just have "locking" and "leasing" be distinct classes of actions w.r.t the wallet.

Reply:
The main gap here is that as is (still need to glance at the latest diff), things like evaluating the balance will examine _leased_ outputs, but not _locked_ outputs.